src/QueryManager.ts

```
@@ -575,6 +584,56 @@ export class QueryManager {
     });
   }
 
+  private collectResultBehaviorsFromReducers(updateQueries, mutationResult, isOptimistic = false) {
+    if (!updateQueries) {
+      return [];
+    }
+    const additionalResultBehaviors = [];
+
+    const observableQueriesByName: { [name: string]: ObservableQuery[] } = {};
+    Object.keys(this.observableQueries).forEach((key) => {
+      const observableQuery = this.observableQueries[key].observableQuery;
+      const queryName = getQueryDefinition(observableQuery.options.query).name.value;
+
+      observableQueriesByName[queryName] =
+        observableQueriesByName[queryName] || [];
+      observableQueriesByName[queryName].push(observableQuery);
+    });
+
+    Object.keys(updateQueries).forEach((queryName) => {
+      const reducer = updateQueries[queryName];
+      const queries = observableQueriesByName[queryName];
+      if (!queries) {
+        // XXX should throw an error?
```

I don't think this should throw an error, because it's pretty reasonable to have some that are not currently on the page... but I agree it could be useful to have something, since people might be confused about why their code is not running.